src/main/java/com/databricks/jdbc/api/impl/LazyThriftResult.java

```
@@ -0,0 +1,256 @@
+          break;
+        }
+
+        // If batch is still empty but hasMoreRows is false after fetch, we'll exit the loop
+      }
+
+      // If we exited the loop and still have an empty batch, we've reached the end
+      if (currentBatch.isEmpty()) {
+        hasReachedEnd = true;
+        globalRowIndex--; // Revert the increment since we didn't actually move to a new row
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Checks if there are more rows available without advancing the cursor.
+   *
+   * @return true if there are more rows, false otherwise
+   */
+  @Override
+  public boolean hasNext() {
+    if (isClosed || hasReachedEnd) {
+      return false;
+    }
+
+    // Check maxRows limit
+    boolean hasRowLimit = maxRows > 0;
+    if (hasRowLimit && globalRowIndex + 1 >= maxRows) {
+      return false;
+    }
+
+    // Check if there are more rows in current batch
+    if (currentBatchIndex + 1 < currentBatch.size()) {
+      return true;
+    }
+
+    // Check if there are more batches to fetch
+    return currentResponse.hasMoreRows;
+  }
+
+  /** Closes this result and releases associated resources. */
+  @Override
+  public void close() {
+    this.isClosed = true;
+    this.currentBatch = null;
+    this.currentResponse = null;
+    LOGGER.debug("LazyThriftResult closed after fetching {} total rows", totalRowsFetched);
+  }
+
+  /**
+   * Gets the number of rows in the current batch.
+   *
+   * @return the number of rows in the current batch
+   */
+  @Override
+  public long getRowCount() {
+    // Return the number of rows in the current batch
+    return currentBatch != null ? currentBatch.size() : 0;
+  }
+
+  /**
+   * Gets the chunk count. Always returns 0 for thrift columnar results.
+   *
+   * @return 0 (thrift results don't use chunks like Arrow)
+   */
+  @Override
+  public long getChunkCount() {
+    // For thrift columnar results, we don't have chunks in the same sense as Arrow
+    return 0;
```

should we instead throw an error since this is an internal interface?